src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -229,6 +229,25 @@ static term create_pair(Context *ctx, term term1, term term2)
     return ret;
 }
 
+static term error_tuple_maybe_gc(Context *ctx, term reason_atom)
+{
+    if (UNLIKELY(!term_is_atom(reason_atom))) {
+        AVM_LOGE(TAG, "Error reason is not an atom! returning 'undefined'");
+        reason_atom = UNDEFINED_ATOM;
+    }
+    if (UNLIKELY(memory_ensure_free(ctx, TUPLE_SIZE(2)) != MEMORY_GC_OK)) {
+        return OUT_OF_MEMORY_ATOM;
+    }
+
+    return create_pair(ctx, ERROR_ATOM, reason_atom);
+}
+
+static term error_tuple_str_maybe_gc(Context *ctx, const char *const reason_str)
```

Comment:
I realize we are going back and forth on these minutia, which is getting kind of silly, but I wonder if you could use the AtomString typedef in atom.h for the type of reason_str?  I haven’t tried it, but maybe the types are incompatible…. If not, please ignore this comment.

Reply:
I don't mind spending the time to get the details right here. Using the `AtomString` typedef here does make a lot more sense. Thank you for the suggestion.

Reply:
Fixed.